rasa/core/policies/intent_ted_policy.py

```diff
@@ -397,26 +403,70 @@ def run_training(
         self.compute_label_quantiles_post_training(model_data, label_ids)
 
     def _collect_action_metadata(
-        self, domain: Domain, similarities: np.array
-    ) -> Dict[Text, Dict[Text, float]]:
+        self, domain: Domain, similarities: np.array, query_intent: Text
+    ) -> Dict[Text, Any]:
         """Adds any metadata to be attached to the predicted action.
 
-        Similarities for all intents and their thresholds are attached as metadata.
+        Metadata schema looks like this:
+
+        {
+            "query_intent": <metadata of intent that was queried>,
+            "ranking": <sorted list of metadata corresponding to all intents.
+                        Sorting is based on predicted similarities.>
+        }
+
+        Each metadata dictionary looks like this:
+
+        {
+            "name": <name of intent>,
+            "score": <predicted similarity score>,
+            "threshold": <threshold used for intent>,
+            "severity": <absolute difference between score and threshold>
```

Comment:
Does severity need to be here if we are additionally storing the numbers to compute it? Not sure what our policy is on the tradeoff cost for storing this info vs recomputing later.

Reply:
Computation of severity is a property of `IntentTEDPolicy`, so it should definitely stay inside it and added to the metadata.
On the tradeoff: Users can always use `ranking_length` parameter to reduce the ranking list if they see that exploding the memory.

Reply:
Thanks for bringing that up, support for `ranking_length` was missing and I added it now.